src/layout/LensInput.js

```diff
@@ -1,10 +1,14 @@
 import React from 'react'
+import _ from 'lodash/fp'
 import { observer } from 'mobx-react'
 import F from 'futil-js'
+import { withTheme } from '../utils/theme'
 
-let LensInput = observer(({ lens, Input, ...x }) => (
-  <Input {...F.domLens.value(lens)} {...x} />
-))
-LensInput.displayName = 'LensInput'
+let LensInput = ({ lens, theme: { Input = 'input' }, ...props }) => (
+  <Input {...F.domLens.value(lens)} {...props} />
+)
 
-export default LensInput
+export default _.flow(
+  observer,
+  withTheme('LensInput')
```

Comment:
Let's be conservative with adding these names. There's no reason I can think of where we'd need to have a different "input" just for LensInput. We can always add these later, but once they're in, changing or removing them is a breaking change